File: 20190309腾讯提前批/1.cpp
#include<string>
#include<vector>
#include<iostream>
#include<algorithm>
#include<stack>
#include<math.h>
#include<map>
#include<set>
using namespace std;

int huan(vector<int>&vec, int s, int m,int& suoyin)
{
	if (vec.size() == 1)
		return vec[0];
	suoyin = (s + 1 + m) % vec.size();
	int result = vec[suoyin];
	vec.erase(vec.begin() + suoyin);
	suoyin = suoyin%vec.size();
	return result;
}

int main() {
	//test1
	int n, s, m;
	cin >> n >> s >> m;
	vector<int>vec(n);
	vector<int>res;
	for (int i = 0; i < n; ++i){
		vec[i] = i + 1;
	}
	int count = n;
	
	int suoyin = s;
	for (int i = 0; i < n; ++i)
	{
		int secondsuoyin;
		int temp = huan(vec, suoyin, m, secondsuoyin);
		res.push_back(temp);
	}
	for (int i = 0; i < res.size(); ++i)
	{
		cout << res[i] << endl;
	}
	//test end
	return 0;
}

File: 20190904华为笔试/1.js
function test(n){
    let count=10000;
    for(let i=0;i<=Math.ceil(n/9);++i){
        for(let j=0;j<=Math.ceil(n/4);++j){
            if(i*9+j*4==n&&i+j<count){
                count=i+j;
            }
        }
    }
    return count==10000?-1:count;

}

function test1(n){
    let num9=Math.floor(n/9);
    for(let i=num9;i>0;--i){
        let left=n-i*9;
        if(left%4==0){
            return i+Math.floor(left/4);
        }
    }
    return -1;
}

console.log(test(13));